CRM/Contribute/Form/Task/Invoice.php

```diff
@@ -413,6 +421,34 @@ public static function printPDF($contribIDs, &$params, $contactIds) {
 
       // from email address
       $fromEmailAddress = $params['from_email_address'] ?? NULL;
+      if (!empty($params['cc_id'])) {
+        // get contacts and their emails from email id
+        $emailIDs = $params['cc_id'] ? explode(',', $params['cc_id']) : [];
+        $emails = Email::get()
+          ->addWhere('id', 'IN', $emailIDs)
+          ->setCheckPermissions(FALSE)
```

Comment:
@sunilpawar I think we don't need setCheckPermissions(FALSE) here - it would make sense to check permissions would't it - the entity ref field has applied permissions

If we do need it then maybe a code comment

Reply:
@eileenmcnaughton i used code from `getEmailString` in `CRM_Contact_Form_Task_EmailTrait` file. I can not use that function directly.

Reply:
ok